### DISCUSSION TITLE
Switch control on Windows vs. other platforms

### DISCUSSION THREAD

**Discussion by kj4bix** [dotnet/maui#17307] [General] [answered]
I'm having an issue with the switch on Windows vs. other platforms. As you can see in the attached image the switch control is far wider on Windows than it is on other platforms. The only way I can get the switch to look correctly on Windows is to add a negative margin (which I was a little surprised worked), but of course that isn't necessary on iOS or Android. I also tried an idiom for the margin on desktop environments (assuming Windows), but get an error that the switch doesn't allow direct context. Does that make sense? I haven't seen that error in the past. Attached is a snapshot from Windows and from Android where it looks ok. Ideas?
<img width="946" alt="windows switch" src="https://github.com/dotnet/maui/assets/9199409/48992221-e866-4e68-bf08-17288f3dbdeb">
<img width="519" alt="android switch" src="https://github.com/dotnet/maui/assets/9199409/1cf2e536-c9c6-44ff-9b5e-badbfca8d23d">

**Comment by Stedy59** [comment, marked as answer]
Put this in your App.xaml located in the Platforms\Windows folder...

`
	<maui:MauiWinUIApplication.Resources>
		<ResourceDictionary>
			
			<Style TargetType="ToggleSwitch">
				<Setter Property="OffContent" Value="      " />
				<Setter Property="OnContent" Value="      " />
				<Setter Property="Margin" Value="0,0,-110,0" />
			</Style>
		</ResourceDictionary>
	</maui:MauiWinUIApplication.Resources>
`

**Reply by kj4bix** [reply]
This works and just impacts Windows which is what I want. I had figured out the negative margin corrects the issue (never used that before).  I don't know why the width of the control is so far off on Windows, but this will correct it for now.

Thanks
Alan

**Reply by Stedy59** [reply]
You're welcome... Glad I could help.
